server/query_test.go

```diff
@@ -137,9 +137,49 @@ func TestParseQuery(t *testing.T) {
 		}
 	}
 
-	_, err := ParseQuery(`hello case:foo`)
-	if err == nil {
-		t.Errorf("parse multiple regexes, no error")
+	lit_cases := []struct {
+		in  string
+		out pb.Query
+	}{
+		{
+			"a( file:b",
+			pb.Query{Line: `a\(`, File: "b", FoldCase: true},
+		},
+		{
+			"a (file:b",
+			pb.Query{Line: `a \(file:b`, FoldCase: true},
+		},
+		{
+			"(file:a b",
```

Comment:
It's not completely clear to me what the "right" behaviour for these cases should be.

Reply:
I think the fact that it's unclear means that probably people won't generate them, and as long as we do something plausible (possibly we should even error?) it's fine.

Alternately, I think that if the "not regex" flag is checked, erring on the side of treating the entire box as a single literal search is pretty fair (which is what this seems to mostly do).

Reply:
I don't think treating the entire box as a single literal search is right. It's still pretty common to want to filter by path. Maybe completely disregarding parentheses is the thing to do?

Reply:
tbc I wasn't suggesting a behavior change; The choices you've made seem defensible, I was just musing.

Reply:
Happy to muse!

I think I may be leaning towards making parentheses completely un-special in literal mode. It seems reasonable to say that if you want to use parens to escape a `file:` or other term, you need to be in regex mode, where parens have a clear and specific meaning. I think that will make the query parser a bit simpler, too.